diff -ur xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/parser.cxx xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/parser.cxx
--- xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/parser.cxx	2010-04-27 21:31:24.000000000 +0200
+++ xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/parser.cxx	2012-07-11 13:01:05.201228465 +0200
@@ -1274,7 +1274,7 @@
     operator () (SemanticGraph::Path const& x,
                  SemanticGraph::Path const& y) const
     {
-        return x.native_file_string () < y.native_file_string ();
+        return x.native () < y.native ();
     }
   };
 
@@ -1627,8 +1627,8 @@
       friend Boolean
       operator< (SchemaId const& x, SchemaId const& y)
       {
-        return x.path_.native_file_string () < y.path_.native_file_string ()
-          || (x.path_.native_file_string () == y.path_.native_file_string ()
+        return x.path_.native () < y.path_.native ()
+          || (x.path_.native () == y.path_.native ()
               && x.ns_ < y.ns_);
       }
 
@@ -4730,7 +4730,7 @@
           ctx_ (ctx)
     {
       setSystemId (XML::XMLChString (
-                     String (abs_.native_file_string ())).c_str ());
+                     String (abs_.native ())).c_str ());
     }
 
     virtual Xerces::BinInputStream*
diff -ur xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx
--- xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx	2010-04-27 21:31:24.000000000 +0200
+++ xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/semantic-graph/elements.cxx	2012-07-11 13:01:05.201228465 +0200
@@ -342,5 +342,5 @@
 std::wostream&
 operator<< (std::wostream& os, XSDFrontend::SemanticGraph::Path const& path)
 {
-  return os << path.native_file_string ().c_str ();
+  return os << path.native ().c_str ();
 }
diff -ur xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/transformations/anonymous.cxx xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/transformations/anonymous.cxx
--- xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/transformations/anonymous.cxx	2010-04-27 21:31:24.000000000 +0200
+++ xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/transformations/anonymous.cxx	2012-07-11 13:01:05.201228465 +0200
@@ -275,7 +275,7 @@
             }
             catch (SemanticGraph::InvalidPath const&)
             {
-              file_str = file.native_file_string ();
+              file_str = file.native ();
             }
 
             String name (
@@ -358,7 +358,7 @@
                 }
                 catch (SemanticGraph::InvalidPath const&)
                 {
-                  file_str = file.native_file_string ();
+                  file_str = file.native ();
                 }
               }
 
@@ -441,7 +441,7 @@
             }
             catch (SemanticGraph::InvalidPath const&)
             {
-              file_str = file.native_file_string ();
+              file_str = file.native ();
             }
 
             String name (
@@ -639,7 +639,7 @@
         }
         catch (SemanticGraph::InvalidPath const&)
         {
-          file_str = file.native_file_string ();
+          file_str = file.native ();
         }
 
         String name (
diff -ur xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/transformations/schema-per-type.cxx xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/transformations/schema-per-type.cxx
--- xsd-3.3.0+dep.orig/libxsd-frontend/xsd-frontend/transformations/schema-per-type.cxx	2010-04-27 21:31:24.000000000 +0200
+++ xsd-3.3.0+dep/libxsd-frontend/xsd-frontend/transformations/schema-per-type.cxx	2012-07-11 13:01:05.201228465 +0200
@@ -358,7 +358,7 @@
         }
         catch (SemanticGraph::InvalidPath const&)
         {
-          abs_path = path.native_file_string ();
+          abs_path = path.native ();
         }
 
         NarrowString tf (trans_.translate_schema (abs_path));
diff -ur xsd-3.3.0+dep.orig/xsd/xsd/cxx/elements.cxx xsd-3.3.0+dep/xsd/xsd/cxx/elements.cxx
--- xsd-3.3.0+dep.orig/xsd/xsd/cxx/elements.cxx	2010-04-28 08:58:09.000000000 +0200
+++ xsd-3.3.0+dep/xsd/xsd/cxx/elements.cxx	2012-07-11 13:01:05.207895131 +0200
@@ -326,7 +326,7 @@
         }
         catch (SemanticGraph::InvalidPath const&)
         {
-          pair = path.native_file_string ();
+          pair = path.native ();
         }
       }
 
diff -ur xsd-3.3.0+dep.orig/xsd/xsd/cxx/parser/elements.cxx xsd-3.3.0+dep/xsd/xsd/cxx/parser/elements.cxx
--- xsd-3.3.0+dep.orig/xsd/xsd/cxx/parser/elements.cxx	2010-04-28 08:58:09.000000000 +0200
+++ xsd-3.3.0+dep/xsd/xsd/cxx/parser/elements.cxx	2012-07-11 13:01:05.207895131 +0200
@@ -244,7 +244,7 @@
       }
       catch (SemanticGraph::InvalidPath const&)
       {
-        path_str = path.native_file_string ();
+        path_str = path.native ();
       }
 
       String inc_path;
diff -ur xsd-3.3.0+dep.orig/xsd/xsd/cxx/parser/generator.cxx xsd-3.3.0+dep/xsd/xsd/cxx/parser/generator.cxx
--- xsd-3.3.0+dep.orig/xsd/xsd/cxx/parser/generator.cxx	2010-04-28 08:58:09.000000000 +0200
+++ xsd-3.3.0+dep/xsd/xsd/cxx/parser/generator.cxx	2012-07-11 13:01:05.207895131 +0200
@@ -638,7 +638,7 @@
       {
         if (NarrowString name = ops.value<CLI::extern_xml_schema> ())
         {
-          if (file_path.native_file_string () != name)
+          if (file_path.native () != name)
             generate_xml_schema = false;
         }
       }
@@ -1010,7 +1010,7 @@
         }
 
         unlinks.add (hxx_impl_path);
-        file_list.push_back (hxx_impl_path.native_file_string ());
+        file_list.push_back (hxx_impl_path.native ());
 
         if (!ops.value<CLI::force_overwrite> ())
         {
@@ -1036,7 +1036,7 @@
         }
 
         unlinks.add (cxx_impl_path);
-        file_list.push_back (cxx_impl_path.native_file_string ());
+        file_list.push_back (cxx_impl_path.native ());
       }
 
       if (driver)
@@ -1065,7 +1065,7 @@
         }
 
         unlinks.add (cxx_driver_path);
-        file_list.push_back (cxx_driver_path.native_file_string ());
+        file_list.push_back (cxx_driver_path.native ());
       }
 
       // Open the skel files.
@@ -1081,7 +1081,7 @@
       }
 
       unlinks.add (hxx_path);
-      file_list.push_back (hxx_path.native_file_string ());
+      file_list.push_back (hxx_path.native ());
 
       if (inline_)
       {
@@ -1094,7 +1094,7 @@
         }
 
         unlinks.add (ixx_path);
-        file_list.push_back (ixx_path.native_file_string ());
+        file_list.push_back (ixx_path.native ());
       }
 
 
@@ -1109,7 +1109,7 @@
         }
 
         unlinks.add (cxx_path);
-        file_list.push_back (cxx_path.native_file_string ());
+        file_list.push_back (cxx_path.native ());
       }
 
       // Print copyright and license.
diff -ur xsd-3.3.0+dep.orig/xsd/xsd/cxx/tree/elements.cxx xsd-3.3.0+dep/xsd/xsd/cxx/tree/elements.cxx
--- xsd-3.3.0+dep.orig/xsd/xsd/cxx/tree/elements.cxx	2010-04-28 08:58:09.000000000 +0200
+++ xsd-3.3.0+dep/xsd/xsd/cxx/tree/elements.cxx	2012-07-11 13:01:05.204561798 +0200
@@ -1312,7 +1312,7 @@
       }
       catch (SemanticGraph::InvalidPath const&)
       {
-        path_str = path.native_file_string ();
+        path_str = path.native ();
       }
 
       String inc_path;
diff -ur xsd-3.3.0+dep.orig/xsd/xsd/cxx/tree/generator.cxx xsd-3.3.0+dep/xsd/xsd/cxx/tree/generator.cxx
--- xsd-3.3.0+dep.orig/xsd/xsd/cxx/tree/generator.cxx	2010-04-28 08:58:09.000000000 +0200
+++ xsd-3.3.0+dep/xsd/xsd/cxx/tree/generator.cxx	2012-07-11 13:01:05.204561798 +0200
@@ -939,7 +939,7 @@
       {
         if (NarrowString name = ops.value<CLI::extern_xml_schema> ())
         {
-          if (file_path.native_file_string () != name)
+          if (file_path.native () != name)
             generate_xml_schema = false;
         }
       }
@@ -1108,7 +1108,7 @@
         }
 
         unlinks.add (fwd_path);
-        file_list.push_back (fwd_path.native_file_string ());
+        file_list.push_back (fwd_path.native ());
       }
 
 
@@ -1121,7 +1121,7 @@
       }
 
       unlinks.add (hxx_path);
-      file_list.push_back (hxx_path.native_file_string ());
+      file_list.push_back (hxx_path.native ());
 
 
       // IXX
@@ -1137,7 +1137,7 @@
         }
 
         unlinks.add (ixx_path);
-        file_list.push_back (ixx_path.native_file_string ());
+        file_list.push_back (ixx_path.native ());
       }
 
 
@@ -1158,7 +1158,7 @@
           }
 
           unlinks.add (*i);
-          file_list.push_back (i->native_file_string ());
+          file_list.push_back (i->native ());
           cxx.push_back (s);
         }
       }
diff -ur xsd-3.3.0+dep.orig/xsd/xsd/xsd.cxx xsd-3.3.0+dep/xsd/xsd/xsd.cxx
--- xsd-3.3.0+dep.orig/xsd/xsd/xsd.cxx	2010-04-28 08:58:09.000000000 +0200
+++ xsd-3.3.0+dep/xsd/xsd/xsd.cxx	2012-07-11 13:01:05.204561798 +0200
@@ -675,7 +675,7 @@
               if (NarrowString name =
                   tree_ops->value<CXX::Tree::CLI::extern_xml_schema> ())
               {
-                if (tu.native_file_string () != name)
+                if (tu.native () != name)
                   gen_xml_schema = false;
               }
             }
@@ -690,7 +690,7 @@
               if (NarrowString name =
                   parser_ops->value<CXX::Parser::CLI::extern_xml_schema> ())
               {
-                if (tu.native_file_string () != name)
+                if (tu.native () != name)
                   gen_xml_schema = false;
               }
             }
diff -ur xsd-3.3.0+dep.orig/xsd/xsd/xsd.hxx xsd-3.3.0+dep/xsd/xsd/xsd.hxx
--- xsd-3.3.0+dep.orig/xsd/xsd/xsd.hxx	2010-04-28 08:58:09.000000000 +0200
+++ xsd-3.3.0+dep/xsd/xsd/xsd.hxx	2012-07-11 13:01:05.201228465 +0200
@@ -38,7 +38,7 @@
   {
     if (!canceled_)
     {
-      std::remove (file_.native_file_string ().c_str ());
+      std::remove (file_.native ().c_str ());
     }
   }
 
